src/SourceBuild/content/eng/tools/BinaryDetection/BinaryDetection/BinaryDetection.csproj

```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <Nullable>enable</Nullable>
+    <SelfContained>true</SelfContained>
+    <PublishTrimmed>true</PublishTrimmed>
+    <NoWarn>$(NoWarn);NU1604</NoWarn>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- There are no defined version props for FileSystemGlobbing, Logging.Console, and Logging in the context of vmr/eng. -->
+    <PackageReference Include="Microsoft.Extensions.FileSystemGlobbing" Version="9.0.0-preview.1.24080.9" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="9.0.0-preview.1.24080.9" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="9.0.0-preview.1.24080.9" />
```

Comment:
I think these package versions will need to be provided at runtime based on the .NET SDK that is being used. So it will either be the case that the prep script installs the SDK or an SDK is provided. Whichever SDK is used, the script will need to call `dotnet --list-runtimes` on it to get the runtime version. That version can be passed as an MSBuild property which would then be consumed here.

That solves the versioning problem. The next would be getting access to the NuGet packages. There would need to be a NuGet.config file defined which has a package feed set to the artifacts path. Again, the prep script would handle the extraction of packages to that path or they'll be explicitly provided. In either case, that path needs to be set as the feed location in the NuGet.config. I know that NuGet.config can handle references to environment variables, but I don't know if that applies to the feed location. If so, then it would be good to define the NuGet.config statically with a variable reference that can be dynamically set. If not, the NuGet.config content would need to be generated dynamically.